src/wp-includes/rest-api.php

```diff
@@ -56,6 +57,11 @@ function register_rest_route( $namespace, $route, $args = array(), $override = f
 		);
 	}
 
+	if ( isset( $args['permissions_callback'] ) ) {
+		_doing_it_wrong( 'register_rest_route', __( 'You have misspelled the permission_callback option.' ), '5.5.0' );
+		return false;
```

Comment:
We don't want to return false here as that'd cause previously working code to break.